### PR TITLE
Refactor/v1 0 0

### LINKED ISSUE #8: Fix wheel speed absolute values
Some environments do not check the absolute value of wheel speeds for failure conditions, and instead just look at the values themselves. This may result in missed constraint violations for large negative wheel speeds. It's worth mentioning that the coupling between power consumption and wheel speeds likely reduces the impact of this bug, but it should be corrected to ensure all envs work as intended.

### LINKED ISSUE #9: Properly disable trackingErrorTask
Some environments may not disable the trackingErrorTask in modes that use attitude guidance modules. Check these environments and correct this issue.

### LINKED ISSUE #10: Refactor environment resource failure print statements
In response to @Mark2000's comment:
"would be cleaner to use f-strings, e.g. print(f"Spacecraft {idx} died from wheel explosion. RPMs were: {observation[8:11]}, limit is {6000 * mc.RPM}, body rate was {observation[7])}. Action taken was {action[idx]}, env step {self.curr_step}.")"

Refactor environment resource constraint failure print statements to use formatted string literals. Refactor for the following environments:
- AgileEOS
- SimpleEOS
- MultiSensorEOS
- MultiSatAgileEOS

### LINKED ISSUE #11: Correct Earth collision check in Earth-observing envs
In response to @Mark2000's comment for the following line of code:

`            if np.linalg.norm(inertialPos) < (orbitalMotion.REQ_EARTH / 1000.0):
`
Change to `orbitalMotion.REQ_EARTH * 1000.0` in MultiSatAgileEOS bsk_sim.

Check that this is correct in other environments.

### LINKED ISSUE #14: Determine release schedule
Discussed paradigm is contribute to develop, then occasionally pull changes on develop into main (once they've been vetted through use). What should this process/schedule look like? Does using github's release features add value?

Should also update version in setup accordingly (https://semver.org).

### LINKED ISSUE #16: Refactor specific environments to use general environment components
Specific environments can be reproduced with the general environment, probably without much additional code beyond configuration. Create and test generic environments against existing ones. Some refactoring of `envs` directory may be useful at this point to reflect the components/instantiation divide.

Considerations
- (How) Should these be importable?
- Assume there will still be types of envs other than the generalized env in the envs directory (e.g. RPOD, EDL), so no deleting the envs layer directory
- Maybe these should be in a subdirectory of general_satellite_tasking? With a nice way of importing/registering/documenting.

Closes #8 #9 #10 #11

### LINKED ISSUE #17: Public-facing documentation
Goal: set up Sphinx docs for the repo

- [X] Use a basilisk docs-like style
- [X] Figure out best deployment strategy (support releases #14 when we start doing that) and location (GitHub? Dr. Schaub's website?). More automated = better
- [X] Primary scope is generalized environment (`bsk_rl/envs/general_satellite_tasking`). Should be optimized for this (e.g. ingest Python type hints, which may make #78 unnecessary). Since other envs are going to be wrapped into the general env (#16), documentation is less important for them; instead consider a good way of documenting some baseline environments (which may include an associated refactor of where those examples should be)
- [ ] The examples directory should be handled like examples in basilisk andother repos (rich formatted walkthrough of the code). 
- [ ] Make changelong page
- [ ] Identify pages that need content written (home pages, examples, etc.)
- [x] Make bsk setters not private so that configs are documented, preferably expose defaults. Maybe rework system to have defaults in constructor and only decorate with names of fields?

If #79 would greatly change configuration of documentation, maybe we should do it first?

### LINKED ISSUE #19: Make vizInterface optional
Vizard imports should be optional so that BSK installs without vizInterface compiled can still be used with the repo

### LINKED ISSUE #55: Use numpy tests in unit tests
Upgrade unit tests to use https://numpy.org/doc/stable/reference/routines.testing.html

### LINKED ISSUE #70: _refine_window raises ValueError
**Describe the bug**
_refine_window raises value error (rarely). Saw it once over 207 days of simulation time. Potentially related to increased default `access_dist_threshold` from 1e6 to 4e6 from #68 or related bug fix in commit `aab0a8a4fb3de0f2e1e4581b075a88617fd71599`.

**To Reproduce**
Unknown

**Expected behavior**
Shouldn't be reachable code.

**Version (please complete the following information):**
 - Python: 3.10.11
 - Basilisk: 2.2.1a
 - Platform: MacOS 13.3

### LINKED ISSUE #80: Standardize environment object construction
Some data managers, communicators, etc are inconsistent in how they handle initialization; some explicitly require other objects to be passed, other can automatically set the object. This should be standardized such that passing other environment construction objects is deprecated in favor of finding them after the environment is constructed.

### LINKED ISSUE #82: Potential for member variable conflicts with satellite actions and observations
If two sat actions/observations share a member variable name, things are gonna break. We should add some way of identifying if that happens, or better yet make it architecturally impossible. Is the current inheritance way of doing things bad?

For observations, a list of some observation type could be passed to the satellite instead of inheriting observations. However, for actions this wouldn't work cause we want to be able to support things other than the discrete action space.

### LINKED ISSUE #115: Make a logo
@LorenzzoQM and I were discussing that it would be nice to have a logo for the package to use in documentation and presentations. Perhaps a robotic looking version of the Basilisk logo? Maybe something different on the shield that is more RL related? @schaubh, do you have the vector graphics file of the basilisk logo somewhere?

### PR DESCRIPTION
**Pull request by Mark2000**
## Description
Closes #115 
Closes #82 
Closes #80 
Closes #70
Closes #55
Closes #19
Closes #17
Closes #16
Closes #14
Closes #11
Closes #10
Closes #9
Closes #8

In progress.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### How should this pull request be reviewed?
- [ ] By commit
- [x] All changes at once

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

### Passes Tests
- [x] __Unit tests__ (General Environment only) `pytest --cov bsk_rl/envs/general_satellite_tasking --cov-report term-missing tests/unittest`
- [x] __Integrated tests__ (General Environment only) `pytest --cov bsk_rl/envs/general_satellite_tasking --cov-report term-missing tests/integration`
- [x] __Examples__ (General Environment only) `pytest tests/examples`

### Test Configuration
- Python:
- Basilisk:
- Platform: 

# Checklist:

- [x] My code follows the style guidelines of this project (passes Black, ruff, and isort)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Commit messages are atomic, are in the form `Issue #XXX: Message` and have a useful message
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Mark2000** [comment]
Will squash changes later, anticipating adding more PRs on review.

**Comment by Mark2000** [comment]
Also looking for opinions on what to do with the deprecated folder.

**Comment by Mark2000** [comment]
Note to self: Update pages build pipeline, remove version warning on index.rst, and release notes date just before merging.

File: src/bsk_rl/obs/observations.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+            return self.obs_list()
+        else:
+            raise ValueError(f"Invalid observation type: {self.obs_type}")
+
+    @property
+    def observation_space(self) -> spaces.Space:
+        """Space of the observation."""
+        obs = self.get_obs()
+        if isinstance(obs, (list, np.ndarray)):
+            return spaces.Box(low=-1e16, high=1e16, shape=obs.shape, dtype=np.float64)
```
Comment (LorenzzoQM) [inline src/bsk_rl/obs/observations.py]: Can you call `obs_dict_to_space(obs)` here as you did in line 127 considering that `obs_dict_to_space` check for types?

Comment (Mark2000) [reply src/bsk_rl/obs/observations.py]: good observation (pun intended)

File: src/bsk_rl/sats/satellite.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        Returns:
+            Satellite's FSW model
+
+        :meta private:
+        """
+        fsw = self.fsw_type(self, fsw_rate, **self.sat_args)
+        self.fsw = proxy(fsw)
+        return fsw
+
+    def reset_post_sim(self) -> None:
```
Comment (LorenzzoQM) [inline src/bsk_rl/sats/satellite.py]: I wonder if changing this to something such as `reste_post_sim_init` could make it more explicit. But now we have the documentation explaining when it is called, so not so necessary

**Comment by LorenzzoQM** [comment]
About the deprecated folder, I don't think we need to keep it in this release, especially considering that some of the code in there might require some changes to work with the updated repo.

**Review by LorenzzoQM — APPROVED** [review]
All excellent changes! Improved usability, and the examples in python notebooks make it easier to learn how to use it.

**Review by Yume27 — APPROVED** [review]
I installed it, ran tests, and ran jupyter notebook scripts. All works great and the jupyter notebook scripts would really help understand the code.

File: src/bsk_rl/sats/satellite.py (hunk 2 above)
Comment (Mark2000) [reply src/bsk_rl/sats/satellite.py]: Still debating whether this is a change worth making. I think docs describe it pretty well and it's rarely something a front end user would touch.

Comment (Mark2000) [reply src/bsk_rl/sats/satellite.py]: Decided to go with your suggestion. Will push that

**Pull request merged**